lib/Net/SAML2/SP.pm

```diff
@@ -303,7 +303,7 @@ sub logout_request {
         issuer        => $self->id,
         destination   => $destination,
         nameid        => $nameid,
-        nameid_format => $nameid_format,
+        $nameid_format ? (nameid_format => $nameid_format) : (),
```

Comment:
I'm ok with this, but with Moose types you can do tricks like this:

NonEmptySimpleStr->check($nameid_format)

This will tell you if it passes the check. Since the name ID format has a type constraint of NonEmptySimpleStr the check is good.
Another option would be an around buildargs for all the options that have the same limitation.

However, I would perhaps change line 138 as well, we use nameid_format there and I dunno what Format="" would do in the XML as an attribute for the nameid.

I have something in my devel branch to address this, but I haven't fully tested it yet.